internal/vfs/vfstest/vfstest.go

```diff
@@ -168,6 +168,29 @@ func (m *mapFS) open(p canonicalPath) (fs.File, error) {
 	return m.m.Open(string(p))
 }
 
+func (m *mapFS) remove(path string) error {
+	canonical := m.getCanonicalPath(path)
+	canonicalString := string(canonical)
+	fileInfo, e := m.m.Stat(canonicalString)
+
+	if e != nil {
+		return e
+	}
+	delete(m.m, canonicalString)
+	delete(m.symlinks, canonical)
+
+	if fileInfo.IsDir() {
+		for path := range m.m {
+			canonicalPath := m.getCanonicalPath(path)
```

Comment:
No need to call `getCanonicalPath` here; `m.m`'s keys are guaranteed to be `canonicalPath`, it's just that due to Go's type system, we can't have `map[canonicalPath]...` and then convert that to `map[string]...`.

Reply:
We should probably stick a comment on `m` that says it's actually `map[canonicalPath]*fstest.MapFile`